3_javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,226 @@
+
+function saveNote(content, id) {
+    notes.push({ content: content, id: id });
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+
+console.log(notes); // [{content: 'Pick up groceries', id: 1}, {content: 'Do laundry', id: 2}]
+
+// --> Get a note
+function getNote(id) {
+    for (let i = 0; i < notes.length; i++) {
+        if (notes[i].id === id) {
+            return notes[i];
+        }
+    }
+}
+
+const firstNote = getNote(1);
+console.log(firstNote); // {content: 'Pick up groceries', id: 1}
+
+
+// --> Log out notes
+function logOutNotesFormatted() {
+    for (let i = 0; i < notes.length; i++) {
+        console.log(`The note with id: ${notes[i].id}, has the following note text: ${notes[i].content}`);
+    }
+}
+
+logOutNotesFormatted();
+
+
+// --> Unique feature --> Deleting notes by ID
+function deleteNote(id) {
+    for (let i = 0; i < notes.length; i++) {
+        if (notes[i].id === id) {
+            notes.splice(i, 1);
+            break;
+        }
+    }
+}
+
+deleteNote(1);
+console.log(notes); // [{content: 'Do laundry', id: 2}]
+
+
+
+
+
+// CactusIO-interactive (Smart phone usage app) optional
+
+// --> Adding an activity
+const activities = []
+
+function addActivity(date, activity, duration) {
+    let newActivity = {
+        date: date,
+        activity: activity,
+        duration: duration
+    };
+    activities.push(newActivity);
+}
+
+addActivity("23/7-18", "Youtube", 30);
+addActivity("23/7-18", "Instagram", 20);
+addActivity("23/7-18", "Tik-Tok", 28);
+console.log(activities);
+
+// --> Usage limit
+let usageLimit = 0;
+
+function setLimit(limit) {
+    usageLimit = limit;
+}
+
+setLimit(60);
+
+// --> Show my status
+function showStatus() {
+    if (activities.length === 0) {
+        return "Add some activities before calling showStatus";
+    } else {
+        let totalDuration = 0;
+        for (let i = 0; i < activities.length; i++) {
+            totalDuration += activities[i].duration;
+        }
+        if (totalDuration > usageLimit) {
+            return "You have reached your limit, no more smartphoning for you!";
+        } else {
+            return "You have added " + activities.length + " activities. They amount to " + totalDuration + " min. of usage";
+        }
+    }
+}
+
+console.log(showStatus());
+
+// --> Extra feature --> Which activity takes the most time?
+function mostTimeSpent() {
+    if (activities.length === 0) {
+        return "No activities recorded";
+    } else {
+        let maxDuration = activities[0].duration;
+        let maxActivity = activities[0].activity;
+        for (let i = 1; i < activities.length; i++) {
+            if (activities[i].duration > maxDuration) {
+                maxDuration = activities[i].duration;
+                maxActivity = activities[i].activity;
+            }
+        }
+        return "The activity you spent the most time on was " + maxActivity + " with " + maxDuration + " min.";
+    }
+}
+console.log(mostTimeSpent());
```

Comment:
🎉 Your addActivity function is clear and concise, showcasing your solid understanding of JavaScript object manipulation.
🎉 Including a `setLimit` function for customizing usage is a thoughtful touch. This adaptability is a strong feature

🔍 Suggestions:

- Consider adding checks in addActivity to verify all parameters. This will help prevent potential bugs due to unexpected inputs.

- Explore using array methods like reduce in showStatus. It's a clean way to tally up durations and could help streamline your code.

- For the mostTimeSpent function, look into the sort method. This can simplify the process and make the function more efficient.